### PR TITLE
command: Do not count data sources in plan totals

### LINKED ISSUE #7483: Counts are wrong with data sources
(Issue deliberately terse and self-assigned!)

Refreshed data sources appear to be included in the counts, probably the hook needs updating.

Same issue as @clstokes found previously.

### PR DESCRIPTION
**Pull request by phinze**
Fixes #7483

### CODE REVIEW

**Comment by phinze** [comment]
cc @jen20 / @jbardin

File: command/hook_count.go [hunk 1]
```diff
@@ -90,6 +91,11 @@ func (h *CountHook) PostDiff(
 	h.Lock()
 	defer h.Unlock()
 
+	// We don't count enything for data sources
+	if strings.HasPrefix(n.Id, "data.") {
```
Comment (apparentlymart) [inline command/hook_count.go]: Might this have the same problem as discussed in #6823? Not sure if `n.Id` here is a module-local name or one that might potentially begin with `module.foo` for data resources in nested modules.

Comment (phinze) [inline command/hook_count.go]: Good question! I thought the same thing, but from reading [`HumanId()`](https://github.com/hashicorp/terraform/blob/edd67547bc2aa12244fcf1620bcb209ee73e008e/terraform/resource.go#L75-L84) I am fairly certain that `InstanceInfo` keeps `Id` unscoped and maintains `ModulePath []string` separately.

Will verify manually as well here.

Comment (phinze) [inline command/hook_count.go]: Yup manual testing confirms this fix works for data sources in modules. 👍

**Comment by jbardin** [comment]
LGTM if we're sure `.data` is always a prefix

**Comment by jen20** [comment]
LGTM I _think_, but I am somewhat concerned about whether the behaviour here will be different if the resource shim (for backwards compatibilty) is used. I'll check and report back...

**Comment by jen20** [comment]
Yup, data sources using the resource shim appear in counts still - however I think this is correct as they will actually evaluate as resources. So LGTM!

**Pull request merged**